File: captainhcg/go/169-majority-element.go
func majorityElement(nums []int) int {
    ele := nums[0]
    cnt := 1
    for _, e := range nums[1:]{
        if ele == e {
            cnt += 1
        } else if cnt == 0 {
            ele = e
            cnt = 1
        } else {
            cnt -= 1
        }
    }
    return ele
}

File: bwu/array/169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maj, cnt = None, 0
        for n in nums:
            if maj is None:
                maj = n
                cnt = 1
            elif n != maj:
                cnt -= 1
                if cnt == 0:
                    maj = None
            else:
                cnt += 1
                
        return maj
